src/ingredient.rs

```diff
@@ -61,7 +62,13 @@ pub trait Ingredient: Any + std::fmt::Debug + Send + Sync {
         db: &'db dyn Database,
         input: Id,
         revision: Revision,
-    ) -> MaybeChangedAfter;
+    ) -> VerifyResult;
+
+    /// Is the value for `input` in this ingredient marked as possibly a provisional cycle value?
+    fn is_verified_final<'db>(&'db self, db: &'db dyn Database, input: Id) -> bool;
+
+    /// Invoked when the current thread needs to wait for a result for the given `key_index`.
```

Comment:
Would you mind documenting the meaning of the return value. What does `true` indicate?